src/codegen/expression.rs

```diff
@@ -1330,7 +1344,39 @@ fn require(
         // On Solana and Substrate, print the reason, do not abi encoding it
         Target::Solana | Target::Substrate { .. } => {
             if let Some(expr) = expr {
-                cfg.add(vartab, Instr::Print { expr });
+                let expr_vec = args.get(1);
+                let final_string = if let ast::Expression::AllocDynamicBytes {
+                    init: Some(vec),
+                    ..
+                } = expr_vec.as_ref().unwrap()
+                {
+                    String::from_utf8(vec.to_vec()).unwrap()
+                } else {
+                    "".to_string()
```

Comment:
This needs fixing. If the revert argument is an expression, then the print should be constructed using the same method as `"{} require condition fail".format(expr)`. 

```rust
let expr = Expression::FormatString(loc, vec![
(FormatString::StringLiteral, Expression::StringLiteral(loc, "foo.sol:102:4: revert failed with reason: ")), 
(FormatString::Default, expr))
]);

Instr::Print { loc, expr }
```